web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -519,7 +515,7 @@
           asset,
         });
       }
-      dispatch('close');
+      await closeViewer();
```

Comment:
This change fixes the described issue while the other changes just remove an unused event, right?

Reply:
Yes, exactly!